### ISSUE TITLE
Support for credential provided by sts.AssumeRole

### ISSUE THREAD

**Issue by mahdi-hdi** [awslabs/collectd-cloudwatch#50]
Currently, the plugin just uses static `aws_access_key` and `aws_secret_key` credential or keys provided by Ec2 AIM role. It's useful to support credential provided by AWS STS which gives the ability to assume different roles. 
An use-case is when you want to put metrics to Cloudwatch in another AWS account. So, you should create a role with proper policies and then use `aws sts` to assume the role in order to get temporary keys.

**Comment by mkozjak** [comment]
Agree. Another use-case is usage on embedded devices that leverage AWS IoT Core to log all system stdout and stderr pipes before the aws iot client kicks in.
awslabs/amazon-cloudwatch-publisher#2
